![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)
## ***Ejercicio Practico Video Games Sales***
        Poniendo en practica lo aprendido en el curso se FreeCodeCamp "Data Analysis with Python"

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)
#### 
**Dia 1 Orientación e Inspección (El Mapa del Tesoro)**

Tema del Día: Inspección de DataFrames. Repaso profundo de .info(), .describe(), .shape, .columns, .dtypes. Entender qué te dice cada uno sobre la "salud" de tus datos.

Aplicación al Proyecto: Carga el CSV de videojuegos y ejecuta todas las funciones de inspección. En una celda de Markdown, anota tus hallazgos iniciales: ¿Cuántas filas/columnas hay? ¿Qué tipos de datos ves? ¿Hay valores nulos a simple vista?

In [1]:
import pandas as pd

print(pd.__version__)

2.3.1


In [2]:
df_videogames = pd.read_csv('vgsales.csv')

df_videogames.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [3]:
df_videogames.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)
###  Hallazgos de inspección .info()

-  Me dice que para la celdas Year, cuenta con datos NaN, por lo que necesitare una estrategia para limpiarlos y quiza por eso Panda lo toma como un dato float64 cuando deberia ser int64.
-  Me dice que para la celdas Publisher, cuenta con datos NaN, por lo que necesitare una estrategia para limpiarlos

In [4]:
df_videogames.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)
###  Hallazgos de inspección .describe()

-  Me dice que para la celda Global_Sales, cuenta con una desviasion estandar de 1.555028 se presume alta, lo que sugiere presencia de outliers (juegos con ventas masivas que podrian sesgar el promedio)

In [5]:
df_videogames.shape

(16598, 11)

In [6]:
df_videogames.columns

Index(['Rank', 'Name', 'Platform', 'Year', 'Genre', 'Publisher', 'NA_Sales',
       'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales'],
      dtype='object')

In [7]:
df_videogames.dtypes

Rank              int64
Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

Tarea: 

¿Qué comando usarías para ver rápidamente el promedio y la mediana de todas las columnas de ventas?

    -   usaria .mean() para el promedio y .median() para la mediana o simplemente usando .describe() ya que el 50% representa la mediana.

¿Qué comando te diría exactamente cuántos valores no nulos hay en la columna Year?

    -   Puedo tener esta informacion de 2 formas, .info() me dice cuantos valores Not Null existen en la fila y tambien podria usar "df_videogames['Year'].notna().sum()" de esta forma tendria el numero de valores no nulos. 

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)
#### 
**Dia 2**

Tema del Día: Entender a fondo la diferencia y el poder de .loc[], .iloc[] y el filtrado booleano.

Tarea: En celdas nuevas, escribe el código necesario para resolver los siguientes desafíos:

  1.  Crea un nuevo DataFrame llamado juegos_recientes que contenga solo los juegos lanzados después del año 2010.

  2.  Crea un nuevo DataFrame llamado juegos_wii que contenga todos los juegos de la plataforma "Wii".

  3.  Crea un DataFrame final llamado hits_de_nintendo que cumpla dos condiciones a la vez:

        -  Que su Publisher sea "Nintendo". Y que sus ventas en Norteamérica (NA_Sales) sean mayores a 10 millones.


#### Practica

In [26]:
df_videogames['Name'].loc[10]

'Nintendogs'

In [27]:
df_videogames['Name'].loc[5:11]

5                        Tetris
6         New Super Mario Bros.
7                      Wii Play
8     New Super Mario Bros. Wii
9                     Duck Hunt
10                   Nintendogs
11                Mario Kart DS
Name: Name, dtype: object

In [29]:
df_videogames[['Name','Platform','Global_Sales']].loc[5:10]

Name Platform  Global_Sales
5                      Tetris       GB         30.26
6       New Super Mario Bros.       DS         30.01
7                    Wii Play      Wii         29.02
8   New Super Mario Bros. Wii      Wii         28.62
9                   Duck Hunt      NES         28.31
10                 Nintendogs       DS         24.76

In [ ]:
df_videogames.iloc[0]

Rank                     1
Name            Wii Sports
Platform               Wii
Year                2006.0
Genre               Sports
Publisher         Nintendo
NA_Sales             41.49
EU_Sales             29.02
JP_Sales              3.77
Other_Sales           8.46
Global_Sales         82.74
Name: 0, dtype: object

In [35]:
df_videogames.tail()

Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  
16595          0.0          0.01  
16596          0.0          0.01  
16597          0.0          0.01

In [37]:
df_videogames[['Name','Platform','Year']].iloc[16593:16598]

Name Platform    Year
16593                Woody Woodpecker in Crazy Castle 5      GBA  2002.0
16594                     Men in Black II: Alien Escape       GC  2003.0
16595  SCORE International Baja 1000: The Official Game      PS2  2008.0
16596                                        Know How 2       DS  2010.0
16597                                  Spirits & Spells      GBA  2003.0

In [40]:
df_videogames.loc[df_videogames['Genre'] == 'Sports']

Rank                           Name Platform    Year   Genre  \
0          1                     Wii Sports      Wii  2006.0  Sports   
3          4              Wii Sports Resort      Wii  2009.0  Sports   
13        14                        Wii Fit      Wii  2007.0  Sports   
14        15                   Wii Fit Plus      Wii  2009.0  Sports   
77        78                        FIFA 16      PS4  2015.0  Sports   
...      ...                            ...      ...     ...     ...   
16576  16579              Rugby Challenge 3     XOne  2016.0  Sports   
16578  16581  Outdoors Unleashed: Africa 3D      3DS  2011.0  Sports   
16579  16582              PGA European Tour      N64  2000.0  Sports   
16581  16584                      Fit & Fun      Wii  2011.0  Sports   
16587  16590       Mezase!! Tsuri Master DS       DS  2009.0  Sports   

                  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0                  Nintendo     41.49     29.02      3.77         8.46   
3                  Nintendo     15.75     11.01      3.28         2.96   
13                 Nintendo      8.94      8.03      3.60         2.15   
14                 Nintendo      9.09      8.59      2.53         1.79   
77          Electronic Arts      1.11      6.06      0.06         1.26   
...                     ...       ...       ...       ...          ...   
16576  Alternative Software      0.00      0.01      0.00         0.00   
16578               Mastiff      0.01      0.00      0.00         0.00   
16579            Infogrames      0.01      0.00      0.00         0.00   
16581               Unknown      0.00      0.01      0.00         0.00   
16587           Hudson Soft      0.00      0.00      0.01         0.00   

       Global_Sales  
0             82.74  
3             33.00  
13            22.72  
14            22.00  
77             8.49  
...             ...  
16576          0.01  
16578          0.01  
16579          0.01  
16581          0.01  
16587          0.01  

[2346 rows x 11 columns]

![green-divider](https://user-images.githubusercontent.com/7065401/52071924-c003ad80-2562-11e9-8297-1c6595f8a7ff.png)